src/dotnet/commands/RestoringCommand.cs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.DotNet.Tools.MSBuild;
+using Microsoft.DotNet.Tools.Restore;
+
+namespace Microsoft.DotNet.Tools
+{
+    public class RestoringCommand : MSBuildForwardingApp
+    {
+        private bool NoRestore { get; }
+
+        private IEnumerable<string> ArgsToForward { get; }
+
+        private IEnumerable<string> ArgsToForwardToRestore
+        {
+            get
```

Comment:
Again here and in a bunch of places. Properties are supposed to be cheap. Their syntax looks like field access. Expensive properties can get mistakenly called more than once: `if (cmd.ArgsToForwardToRestore.Any()) DoSomething(cmd.ArgsToForwardToRestore)` and it doesn't read like we're repeating work.